components/InputTypeLocation.js

```diff
@@ -151,16 +151,29 @@ class InputTypeLocation extends React.Component {
       return this.props.onChange(location);
     }
 
-    const countryComponent = value.gmaps['address_components'].find(c => c.types.includes('country'));
+    const country = value.gmaps['address_components'].find(c => c.types.includes('country'))?.['short_name'];
+
+    /* Use ADR microformat field `adr_address` because of more consistent formatting and since
+       it also includes a single field for street address (with house number in the correct place depending on locality) */
+    const adrAddress = value.gmaps['adr_address'];
+    const streetAddress = adrAddress.match(/<span class="street-address">([^<]+)<\/span>/i)?.[1];
+    const postalCode = adrAddress.match(/<span class="postal-code">([^<]+)<\/span>/i)?.[1];
+    const city = adrAddress.match(/<span class="locality">([^<]+)<\/span>/i)?.[1];
+    const zone = adrAddress.match(/<span class="region">([^<]+)<\/span>/i)?.[1];
     const location = {
       // Remove country from address
       address: this.removeCountryFromAddress(value.gmaps.formatted_address),
       // Keep only the first part for location name
       name: value.label && value.label.replace(/,.+/, ''),
-      // Normally returned addresses always have a country, this is just defensive
-      country: countryComponent ? countryComponent['short_name'] : null,
+      country,
       lat: value.location.lat,
       long: value.location.lng,
+      structured: {
+        address1: streetAddress,
+        postalCode,
+        city,
+        zone,
```

Comment:
A bit surprised that we build the same structure regardless of the country; e.g. there's often a `address2` field for france

Reply:
I've been assuming that Google Places Autocomplete will never return something akin to a `address2` field, but seeing that there is a `extended-address` field in the adr-microformat I'm now including that as `address`. 

Also added a step to omit any null or undefined values from the `structured` JSON.